torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -838,14 +838,15 @@ at::Tensor& AtenXlaType::copy_(at::Tensor& self, const at::Tensor& src,
     XLA_CHECK(self_tensor);
     self_tensor->UpdateFromTensor(CopyTensor(src, self.scalar_type()));
   } else if (!self_tensor) {
-    // TODO: Is self_tensor good enough?  I don't think so... therefore
-    // the hack below:
     std::vector<at::Tensor> tensors = {src};
     auto xla_tensors = bridge::XlaCreateTensorList(tensors);
-    // Hack in an overwrite of a const tensor.
     at::Tensor t = CopyTensor(xla_tensors.front(), self.scalar_type());
```

Comment:
Our `CopyTensor()` always make a copy (`copy=true`) and here we only need that if `dtype` does not match.
I'd expand the proper `.to()` here.